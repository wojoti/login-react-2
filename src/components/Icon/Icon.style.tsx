import styled from "styled-components";
type Props = {
  w: number;
  h: number;
};
export const Wrapper = styled.img<Props>`
  width: ${(props) => props.w || 0}px;
  height: ${(props) => props.h || 0}px;
`;
