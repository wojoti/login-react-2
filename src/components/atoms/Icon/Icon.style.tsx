import styled from 'styled-components'
interface Props {
  width: number
  height: number
}
export const StyledIcon = styled.img<Props>`
  width: ${(props) => props.width || 0}px;
  height: ${(props) => props.height || 0}px;
`
