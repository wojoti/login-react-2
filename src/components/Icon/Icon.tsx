import React from "react";
import { Wrapper } from "./Icon.style";
export enum IconType {
  facebook = "./img/facebook.svg",
  linkedin = "./img/linkedin.svg",
  google = "./img/google.svg",
}

type Props = {
  type: IconType;
  w: number;
  h: number;
  alt: string;
};

export const Icon = (props: Props) => {
  return (
    <Wrapper src={props.type} w={props.w} h={props.h} alt={props.alt}></Wrapper>
  );
};
