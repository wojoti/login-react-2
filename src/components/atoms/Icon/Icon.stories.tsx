/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Icon, { IconType } from "./Icon";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Icon",
  component: Icon,
} as ComponentMeta<typeof Icon>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} />;

export const Google = Template.bind({});
Google.args = {
  src: IconType.google,
  width: 16,
  height: 16,
  alt: "google",
};

export const Facebook = Template.bind({});
Facebook.args = {
  src: IconType.facebook,
  width: 16,
  height: 16,
  alt: "facebook",
};

export const LinkedIn = Template.bind({});
LinkedIn.args = {
  src: IconType.linkedin,
  width: 16,
  height: 16,
  alt: "linkedin",
};
