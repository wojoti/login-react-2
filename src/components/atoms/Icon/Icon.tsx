import React from 'react'
import { StyledIcon } from './Icon.style'
export enum IconType {
  facebook = './img/facebook.svg',
  linkedin = './img/linkedin.svg',
  google = './img/google.svg',
}

interface Props {
  type: IconType
  width: number
  height: number
  alt: string
}

export const Icon = (props: Props) => {
  return (
    <StyledIcon
      src={props.type}
      width={props.width}
      height={props.height}
      alt={props.alt}
    />
  )
}
