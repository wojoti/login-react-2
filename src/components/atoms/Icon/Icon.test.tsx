import { cleanup, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Icon, { IconProps, IconType } from "./Icon";

afterEach(() => {
  cleanup();
});

const props: IconProps = {
  src: IconType.facebook,
  width: 16,
  height: 16,
  alt: "alt-text",
  testId: "test-icon-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Icon {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render icon", () => {
  render(<Icon {...props} />);
  const iconElement = screen.getByTestId("test-icon-id");
  expect(iconElement).toBeInTheDocument();
});

test("should render icon with alternative name", () => {
  render(<Icon {...props} />);
  const iconElement = screen.getByAltText("alt-text");
  expect(iconElement).toBeInTheDocument();
});
