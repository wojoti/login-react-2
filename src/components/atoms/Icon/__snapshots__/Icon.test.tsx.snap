// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 16px;
  height: 16px;
}

<img
  alt="alt-text"
  className="c0"
  data-testid="test-icon-id"
  height={16}
  src="./img/facebook.svg"
  width={16}
/>
`;
